DEVELOPING.md

```diff
@@ -2,15 +2,62 @@
 
 We strongly encourage you to review the project's scope described in the `README.md` file before working on new features. For large changes, consider writing a design document using [this template](https://goo.gl/YCQttR).
 
-## Testing locally
+## Architecture
 
-This repository contains nested workspaces which are tested with the bazel-in-bazel bazel_integration_test rule. The integration tests must be run in series as they use up too many resources when run in parallel.
+Unlike some other Bazel rules, these rules require a separate build step, and
```

Comment:
This is possible, see the [from_source](https://github.com/bazelbuild/rules_nodejs/tree/stable/examples/from_source) example, I'll agree though it's not intuitive of how to set it up.